src/ucs/async/async.c

```diff
@@ -332,12 +332,18 @@ ucs_status_t ucs_async_dispatch_handlers(int *handler_ids, size_t count,
 ucs_status_t ucs_async_dispatch_timerq(ucs_timer_queue_t *timerq,
                                        ucs_time_t current_time)
 {
-    size_t max_timers, num_timers = 0;
+    size_t num_timers = 0;
+    ucs_status_t status;
+    size_t max_timers;
     int *expired_timers;
     ucs_timer_t *timer;
 
-    max_timers     = ucs_max(1, ucs_timerq_size(timerq));
-    expired_timers = ucs_alloca(max_timers * sizeof(*expired_timers));
+    max_timers     = ucs_min(1024, ucs_max(1, ucs_timerq_size(timerq)));
+    expired_timers = ucs_malloc(max_timers * sizeof(*expired_timers),
```

Comment:
seems it can potentially hurt performance. Is there really the case when 1K is not enough?

Reply:
currently `ucs_alloca()` allows max 1200 bytes, here we are capping at 1024*4 bytes = 4096 bytes so we should not hurt performance, as the limit has grown.

Reply:
i meant malloc overhead

Reply:
I see, yes seems so, also I though we are going away from stack allocation anyways. Do you think we should keep that buffer around, in some context?

Reply:
question is why we have so many timers, AFAIR the UD timer is per iface?
and if really need to support many timers, maybe allocate small size using alloca and large size using malloc

Reply:
Yes, could not find users in UCX and UD transport is disabled, but I suspect we have external app using it, and with this PR, issue is completely gone.

Reply:
fixed